src/api/odometry/odomState.cpp

```diff
@@ -15,11 +15,20 @@ bool OdomState::operator!=(const OdomState &rhs) const {
   return !(rhs == *this);
 }
 
-std::string OdomState::str() const {
-  std::ostringstream os;
-  os << "OdomState(x=" << std::to_string(x.convert(meter))
-     << "m, y=" << std::to_string(y.convert(meter))
-     << "m, theta=" << std::to_string(theta.convert(degree)) << "deg)";
-  return os.str();
+std::string OdomState::str(QLength idistanceUnit,
+                           std::string distUnitName,
+                           QAngle iangleUnit,
+                           std::string angleUnitName) const {
+  char buf[150];
+  snprintf(buf, sizeof(buf), "OdomState(x=%.2f%s, y=%.2f%s, theta=%.2f%s)", x.convert(idistanceUnit), distUnitName.c_str(), y.convert(idistanceUnit), distUnitName.c_str(), theta.convert(iangleUnit), angleUnitName.c_str());
+  return std::string(buf);
```

Comment:
Rewrote because the '<<' syntax was getting messy when the units were also specified. Is this fine? 
A 'normal' string `OdomState(x=2.56m, y=0.80m, theta=69.00deg)` is 44 bytes, so 150 ought to be more than enough for anything.

This is using snprintf because PROS uses c++17 by default, which doesn't support std::format. https://github.com/purduesigbots/pros/issues/375